### DISCUSSION TITLE
Questions regarding local plugin development with python.

### DISCUSSION THREAD

**Discussion by UltraInstinct05** [Flow-Launcher/Flow.Launcher#3006] [Q&A]
Hello 👋 

I am trying to learn how to develop plugins for Flow Launcher using python. Using the documentation (https://www.flowlauncher.com/docs/#/py-develop-plugins), I am trying to get the example plugin to actually work locally on a Windows 11 PC.

Using Flow Launcher, I navigate to the user data folder (`~/AppData/Roaming/FlowLauncher`) and within the `Plugins` folder, I create a folder called `Flow.Launcher.Plugin.HelloWorld` with the following folder structure (I assume this is the method based on https://www.flowlauncher.com/docs/#/testing)

```
.
└── Plugins/
    └── Flow.Launcher.Plugin.HelloWorld/
        ├── main.py
        └── plugin.json
```

I copy the code (as outlined in https://www.flowlauncher.com/docs/#/py-write-code?id=_2-mainpy) to the `main.py` file. 

My `plugin.json` file looks like

```json
{
    "ID":"",
    "ActionKeyword":"hello_world",
    "Name":"Flow.Launcher.Plugin.HelloWorld",
    "Description":"Example Flow Launcher Plugin",
    "Author":"John Doe",
    "Version":"",
    "Language":"python",
    "Website":"",
    "IcoPath": "",
    "ExecuteFileName":"main.py"
}
```

At this point, I restart Flow Launcher and type `hello_world`, but don't see any results that matches what the plugin should have done. At this point, the documentation doesn't seem to have many details on debugging etc.

What am I missing here in order to make the example plugin work ?

**Comment by Garulf** [comment]
Hi @UltraInstinct05,

The `ID` field must not be blank and unique to your plugin. 

A UUID4 is typically used as the ID.

**Reply by UltraInstinct05** [reply]
Hi @Garulf Thanks for the reply. What do people usually use to generate these UUID4 ?

**Reply by UltraInstinct05** [reply]
Also is there some place where the plugin errors are logged ? Like a dev console ?

**Reply by jjw24** [reply]
> Hi @Garulf Thanks for the reply. What do people usually use to generate these UUID4 ?

Any online generator will be able to do so- https://www.uuidgenerator.net/version4

**Reply by jjw24** [reply]
> Also is there some place where the plugin errors are logged ? Like a dev console ?

If you haven't implemented logging yourself then errors will be logged in flow's own logs. Type `open logs`
